src/lambdaisland/witchcraft/citizens.clj

```diff
@@ -0,0 +1,63 @@
+(ns lambdaisland.witchcraft.citizens
+  (:require [lambdaisland.witchcraft :as wc]
+            [lambdaisland.witchcraft.events :as e]
+            [lambdaisland.witchcraft.citizens.trait :as trait]
+            [lambdaisland.witchcraft.reflect :as reflect]
+            [lambdaisland.witchcraft.util :as util])
+  (:import (net.citizensnpcs Citizens)
+           (net.citizensnpcs.api CitizensAPI)
+           (net.citizensnpcs.api.trait Trait TraitFactory)
+           (net.citizensnpcs.api.npc NPCRegistry)
+           (org.bukkit.entity EntityType)))
+
+(defn npc-registry ^NPCRegistry []
+  (CitizensAPI/getNPCRegistry))
+
+(defn trait-factory ^TraitFactory []
+  (CitizensAPI/getTraitFactory))
+
+(defprotocol HasCreateNPC
+  (-create-npc [_ e s]))
+
+(reflect/extend-signatures HasCreateNPC
+                           "createNPC(org.bukkit.entity.EntityType,java.lang.String)"
+                           (-create-npc [this entity-type npc-name]
+                                        (.createNPC this entity-type npc-name)))
+
+(defn create-npc [entity-type npc-name]
+  (-create-npc (npc-registry)
+               (get wc/entity-types entity-type)
+               npc-name))
```

Comment:
Just for my own understanding: Is this strictly necessary here or did you add it because it will come in handy later? `npc-registry` has a type annotation, so I would expect that the Clojure compiler knows that a `createNPC` method with 2 arguments (+ this) exists, so it does not have to use reflection to find it when calling it on `(npc-registry)`.

Reply:
In this case it's not strictly necessary, no.